pkg/skaffold/initializer/build/resolve.go

```diff
@@ -74,14 +83,37 @@ func (d *defaultBuildInitializer) resolveBuilderImagesInteractively() error {
 		}
 
 		if choice != NoBuilder {
-			pairs = append(pairs, BuilderImagePair{Builder: choiceMap[choice], ImageName: image})
+			d.builderImagePairs = append(d.builderImagePairs, BuilderImagePair{Builder: choiceMap[choice], ImageName: image})
 			choices = util.RemoveFromSlice(choices, choice)
 		}
 		d.unresolvedImages = util.RemoveFromSlice(d.unresolvedImages, image)
 	}
 	if len(choices) > 0 {
-		logrus.Warnf("unused builder configs found in repository: %v", choices)
+		// TODO(nkubala): should we ask user if they want to generate here?
+		for _, choice := range choices {
+			d.generatedBuilderImagePairs = append(d.generatedBuilderImagePairs, getGeneratedBuilderPair(choiceMap[choice]))
+		}
 	}
-	d.builderImagePairs = append(d.builderImagePairs, pairs...)
 	return nil
 }
+
+func getGeneratedBuilderPair(b InitBuilder) GeneratedBuilderImagePair {
+	path := b.Path()
+	var imageName string
+	// if the builder is in a nested directory, use that as the image name AND the path to write the manifest
+	// otherwise, use the builder as the image name itself, and the current directory to write the manifest
+	if filepath.Dir(path) != "." {
+		imageName = strings.ToLower(filepath.Dir(path))
+		path = imageName
+	} else {
+		imageName = fmt.Sprintf("%s-image", strings.ToLower(path))
```

Comment:
ok. i see the `path` being part of the imageName.

Reply:
:+1: